eng/codecov.yml

```diff
@@ -0,0 +1,11 @@
+codecov:
+  branch: master
+  ci:
+    - dnceng.visualstudio.com
```

Comment:
❔ Can you provide more information about what this does?

Reply:
Sure: https://docs.codecov.io/docs/detecting-ci-services#section-detection-technique.

As we upload our reports from helix machines, the codecov's detection technique fails and can't figure out which CI service is used. We ideally should set this variable in the teams codecov yml but I don't permissions to do so: https://docs.codecov.io/docs/codecov-yaml#section-team-yaml